### PR TITLE
FIST documentation and examples in discretization.md

### PR DESCRIPTION
**Pull request by atreyamaj**
* Created a section called `Methods` in `discretization.md` so that future discretization methods being developed can be added there seamlessly.
* Shifted `FIST` docstring to `Methods`
* Added example code for `FIST`
* Added further examples for more complex polygons and their corresponding FIST-discretized triangulated meshes, as per serenity4's review.
* Attempted to make the code example easier and more intuitive to understand.
* Fixed spacing and similar formatting issues to the best of my ability.
* Followed the documentation example of `GeoStats.jl`, particularly the `Domains` subsection for how to add images for specific algorithms and have followed it as much as possible, with only slight modifications pertaining to the further examples.
* Ensured that `Plots.jl` is present in `docs/Project.toml`
* Have done my best to follow the document shared for making high-quality PRs.

The page currently looks like this:
Screenshot from 2021-03-28 18-30-38
Screenshot from 2021-03-28 18-30-41
Screenshot from 2021-03-28 18-30-47

I hope this is okay. Please let me know if there are changes to be made.
Thank you.

### CODE REVIEW

File: docs/src/algorithms/discretization.md [hunk 1]
```diff
@@ -3,5 +3,70 @@
...
+
+### FIST
+
+```@docs
 FIST
 ```
+
+```@example FIST
+using Meshes
+using Plots;
```
Comment (juliohm) [inline docs/src/algorithms/discretization.md]:
```suggestion
using Plots
```

File: docs/src/algorithms/discretization.md [hunk 2]
```diff
@@ -3,5 +3,70 @@
...
+using Meshes
+using Plots
+gr(size = (500,300)) # hide
+
+# Outer and inner chains of Polygon
+outer = [(0.22926679, 0.47329807), (0.45326096, 0.2665109), (0.51871038, 0.22148979), (0.3152017, 0.31646582), (0.5953296, 0.62900037), (0.5951828, 0.6113712), (0.74374446, 0.72971905), (0.29018405, 0.5538437), (0.25439468, 0.5678829), (0.22926679, 0.47329807)]
+inner = []
+
+# Construct Polygon
+poly = PolyArea(outer, inner)
```
Comment (juliohm) [inline docs/src/algorithms/discretization.md]:
```suggestion
inners = []

# Construct Polygon
poly = PolyArea(outer, inners)
```

File: docs/src/algorithms/discretization.md [hunk 3]
```diff
@@ -3,5 +3,70 @@
...
+using Meshes
+using Plots
+gr(size = (500,300)) # hide
+
+# Outer and inner chains of Polygon
+outer = [(0.22926679, 0.47329807), (0.45326096, 0.2665109), (0.51871038, 0.22148979), (0.3152017, 0.31646582), (0.5953296, 0.62900037), (0.5951828, 0.6113712), (0.74374446, 0.72971905), (0.29018405, 0.5538437), (0.25439468, 0.5678829), (0.22926679, 0.47329807)]
+inners = []
+
+# Construct Polygon
+poly = PolyArea(outer, inners)
```
Comment (juliohm) [inline docs/src/algorithms/discretization.md]:
```suggestion
# example polygonal area
outer = [(0.22926679, 0.47329807), (0.45326096, 0.2665109), (0.51871038, 0.22148979), (0.3152017, 0.31646582), (0.5953296, 0.62900037), (0.5951828, 0.6113712), (0.74374446, 0.72971905), (0.29018405, 0.5538437), (0.25439468, 0.5678829), (0.22926679, 0.47329807)]
inners = []
polyarea = PolyArea(outer, inners)
```

File: docs/src/algorithms/discretization.md [hunk 4]
```diff
@@ -3,5 +3,70 @@
...
+
+# Discretize with FIST
+mesh = discretize(poly, FIST())
```
Comment (juliohm) [inline docs/src/algorithms/discretization.md]:
```suggestion
mesh = discretize(polyarea, FIST())
```

File: docs/src/algorithms/discretization.md [hunk 5]
```diff
@@ -3,5 +3,68 @@
...
+polyarea = PolyArea(outer, inners)
+
+# Discretize with FIST
+mesh = discretize(polyarea, FIST())
+
+# Plot the polygon and mesh
+p1 = plot(poly)
+p2 = plot(mesh)
+
+plot(p1, p2)
```
Comment (juliohm) [inline docs/src/algorithms/discretization.md]:
```suggestion
# polygonal area
outer = [(0.22926679, 0.47329807), (0.45326096, 0.2665109), (0.51871038, 0.22148979), (0.3152017, 0.31646582), (0.5953296, 0.62900037), (0.5951828, 0.6113712), (0.74374446, 0.72971905), (0.29018405, 0.5538437), (0.25439468, 0.5678829), (0.22926679, 0.47329807)]
inners = []
polyarea = PolyArea(outer, inners)

# FIST discretization
mesh = discretize(polyarea, FIST())

plot(plot(polyarea), plot(mesh))
```

File: docs/src/algorithms/discretization.md [hunk 6]
```diff
@@ -3,5 +3,68 @@
...
+plot(plot(poly), plot(mesh)) # hide
+```
+
+```@example FIST
+poly = PolyArea(Point2f[(0.20974362, 0.11143557), (0.24619843, 0.019633872), (0.2525517, 0.10491638), (0.2605699, 0.03590731), (0.2794662, 0.009585844), (0.54061747, 0.2523168), (0.5210726, 0.27176696), (0.55140936, 0.29682124), (0.5437957, 0.24153952), (0.6285009, 0.21651964), (0.6449456, 0.2147006), (0.59213245, 0.26681182), (0.8229673, 0.74253976), (0.65908766, 0.56879747), (0.5620604, 0.53223926), (0.52954125, 0.5394655), (0.23878936, 0.1011955), (0.22548631, 0.09457563), (0.2759589, 0.71809477), (0.5383939, 0.85454345), (0.6496192, 0.9248424), (0.6248543, 0.8470848), (0.749698, 0.8314047), (0.75198966, 0.9489501), (0.7027816, 0.92791766), (0.7210894, 0.8686051), (0.65642047, 0.9434703), (0.6423689, 0.9363239), (0.549272, 0.9676014), (0.23183312, 0.71278405), (0.20974362, 0.11143557)], []) # hide
+
+mesh = discretize(poly, FIST()) # hide
+
+plot(plot(poly), plot(mesh)) # hide
+```
```
Comment (juliohm) [inline docs/src/algorithms/discretization.md]: I think we only need one example, or perhaps two with different features. One example where polygonal area doesn't have holes (inner rings) and another example with holes.

**Review by juliohm — CHANGES_REQUESTED** [review]
The PR is much improved compared to the last one. Thank you!

I still can't find the update to docs/Project.toml.

**Comment by juliohm** [comment]
Also your first example doesn't seem to be one of the examples in the test suite.

File: docs/src/algorithms/discretization.md (hunk 6 above)
Comment (atreyamaj) [reply docs/src/algorithms/discretization.md]: Thank you, I will work on this immediately.

**Comment by atreyamaj** [comment]
> Also your first example doesn't seem to be one of the examples in the test suite.

I was using the polygon from `smooth1.line` at first, however serenity4 wanted me to use a simpler polygon with around 10 points. Thus I removed quite a few points to convert it into a more simple polygon. Would you like to me use one of the later polygons from among the examples instead?

**Comment by juliohm** [comment]
There are many examples in the test suite with fewer vertices. Also, you cannot remove points arbitrarily. See how your screenshot contains a polygon with an edge crossing another. This is the kind of detail that you should pay attention to before submitting a PR and asking for a review. None of he other examples have this crossing property.

**Comment by atreyamaj** [comment]
> The PR is much improved compared to the last one. Thank you!
> 
> I still can't find the update to docs/Project.toml.

Thank you.
I followed the instructions you shared in the last PR.
First I changed my directory to point inside docs by executing `cd("./docs")`
Next I entered the package manager mode and executed `add Plots` and also immediately afterwards executed `Pkg.add("Plots")` in the Julia terminal within `docs` for good measure. 

`docs/Project.toml` shows a `Plots` entry for me, and it is showing up in the manifest as well.

I will look into rectifying this.

**Comment by atreyamaj** [comment]
> There are many examples in the test suite with fewer vertices. Also, you cannot remove points arbitrarily. See how your screenshot contains a polygon with an edge crossing another. This is the kind of detail that you should pay attention to before submitting a PR and asking for a review. None of he other examples have this crossing property.

I am changing it now. Thank you for pointing this out.

**Comment by juliohm** [comment]
You have to think on your own @atreyamaj. Be the owner of your thoughts. Think critically about everything that is happening around you. That is the path to success. @serenity4 gave you a good suggestion but you have to process it with your own mind first before just pushing a commit with a random deletion of vertices.

**Comment by atreyamaj** [comment]
> You have to think on your own @atreyamaj. Be the owner of your thoughts. Think critically about everything that is happening around you. That is the path to success. @serenity4 gave you a good suggestion but you have to process it with your own mind first before just pushing a commit with a random deletion of vertices.

Thank you. I will remember this and do my best to apply it continuously.

File: docs/src/algorithms/discretization.md [hunk 7]
```diff
@@ -3,5 +3,56 @@
...
+
+# polygonal area
+outer = [(0.16687596, 0.6763949), (0.63792515, 0.20132281), (0.92808944, 0.52010834), (0.95507824, 0.53952533), (0.9225352, 0.5166371), (0.8190737, 0.5040926), (0.7926518, 0.5494572), (0.74116296, 0.5993454), (0.79804873, 0.5519731), (0.85590965, 0.6494151), (0.49462578, 0.93649143), (0.47591516, 0.98907244), (0.46890727, 0.90921813), (0.36027902, 0.99050534), (0.33441278, 0.9391666), (0.16687596, 0.6763949)]
+inners = []
+polyarea = PolyArea(outer, inners)
+
+# FIST discretization
+mesh = discretize(polyarea, FIST())
+
+plot(plot(polyarea), plot(mesh))
```
Comment (juliohm) [inline docs/src/algorithms/discretization.md]:
```suggestion
polyarea = PolyArea([(0.16687596, 0.6763949), (0.63792515, 0.20132281), (0.92808944, 0.52010834), (0.95507824, 0.53952533), (0.9225352, 0.5166371), (0.8190737, 0.5040926), (0.7926518, 0.5494572), (0.74116296, 0.5993454), (0.79804873, 0.5519731), (0.85590965, 0.6494151), (0.49462578, 0.93649143), (0.47591516, 0.98907244), (0.46890727, 0.90921813), (0.36027902, 0.99050534), (0.33441278, 0.9391666), (0.16687596, 0.6763949)])

mesh = discretize(polyarea, FIST())

plot(plot(polyarea), plot(mesh))
```

**Review by juliohm — CHANGES_REQUESTED** [review]
Hi @atreyamaj, thank you for updating the PR. Can you please provide a screenshot of the two polygons you have chosen? Ideally one of them will contain holes to illustrate the advantage of the FIST algorithm.

**Comment by atreyamaj** [comment]
> Hi @atreyamaj, thank you for updating the PR. Can you please provide a screenshot of the two polygons you have chosen? Ideally one of them will contain holes to illustrate the advantage of the FIST algorithm.

Oh I forgot to attach a screenshot earlier, my apologies @juliohm ! 
For the simple (first) polygon, I have taken only the outer area of one of the polygons from `hole.line` as it has very few (10-15) vertices, thus making it a simple hole-less polygon. For the other two example polygons, I have taken a smooth polygon from ` smooth.line` and another with holes from `hole.line`. Here are the attached screenshots:
Screenshot from 2021-03-29 08-48-49

Screenshot from 2021-03-29 08-48-55

**Comment by juliohm** [comment]
@atreyamaj can you please select one smooth example without holes and another example with holes? We only need two examples in this section. In the first example, we can plot the result right away. After the first example we need a short sentence explaining that FIST is also suitable for polygonal areas with holes. Finally, the last example is run with holes to illustrate the feature.

**Comment by atreyamaj** [comment]
> @atreyamaj can you please select one smooth example without holes and another example with holes? We only need two examples in this section. In the first example, we can plot the result right away. After the first example we need a short sentence explaining that FIST is also suitable for polygonal areas with holes. Finally, the last example is run with holes to illustrate the feature.

Apologies for the delay, @juliohm ,I have made the aforementioned changes. The page currently looks like this:
Screenshot from 2021-03-29 21-23-59

I was wondering if maybe adding the code for the second example might be a good idea since it differs from the first one.

**Comment by juliohm** [comment]
We will need to iterate again because you didn't pay attention to what I said (again). I said the following:

>@atreyamaj can you please select one smooth example without holes and another example with holes? We only need two examples in this section. In the first example, we can plot the result right away. After the first example we need a short sentence explaining that FIST is also suitable for polygonal areas with holes. Finally, the last example is run with holes to illustrate the feature.

Your screenshot doesn't seem to contain any smooth example. Can you please share the files you copied from the test suite?

**Comment by atreyamaj** [comment]
> We will need to iterate again because you didn't pay attention to what I said (again). I said the following:
> 
> > @atreyamaj can you please select one smooth example without holes and another example with holes? We only need two examples in this section. In the first example, we can plot the result right away. After the first example we need a short sentence explaining that FIST is also suitable for polygonal areas with holes. Finally, the last example is run with holes to illustrate the feature.
> 
> Your screenshot doesn't seem to contain any smooth example. Can you please share the files you copied from the test suite?

Sorry, I thought when you said 'smooth' you meant a polygon with no holes. I have changed the first example to `smooth3.line`. The second example is `hole3.line`.
Screenshot from 2021-03-29 21-40-32

**Comment by juliohm** [comment]
Let me try it again:

Example 1: A smooth curve with no holes, try to pick one with more area like the one you had in the beginning of the PR:

image

Example 2: A smooth curve with holes. Are there any? In case there are no examples of smooth curves with holes, then pick and example from the holes.line family with more area too. This example you picked is not very easy to understand what is a hole.

**Comment by atreyamaj** [comment]
Updated, please check @juliohm 
There are no `smooth.line` polygons with holes. Replaced hole plot with a plot which has more area.
Screenshot from 2021-03-29 21-56-50

**Comment by juliohm** [comment]
Awesome! The examples look great now! I'll review the details soon...

File: docs/src/algorithms/discretization.md [hunk 8]
```diff
@@ -3,5 +3,41 @@
...
+```@example FIST
+outer = [(0.18142937, 0.54681134), (0.38282228, 0.107781954), (0.43220532, 0.013640274), (0.48068276, 0.019459315), (0.48322055, 0.11583236), (0.46696007, 0.2230227), (0.48184678, 0.2656454), (0.45998818, 0.2784367), (0.4168235, 0.2190962), (0.4124987, 0.21208182), (0.39593673, 0.2520411), (0.44333926, 0.28375763), (0.4978224, 0.3981428), (0.7703431, 0.20181546), (0.7612364, 0.33008572), (0.9856581, 0.2215304), (0.99374324, 0.3353423), (0.9688778, 0.38663587), (0.59554976, 0.655444), (0.59496254, 0.58492756), (0.27641845, 0.656314), (0.3242084, 0.6072907), (0.42408508, 0.49353212), (0.20984341, 0.59003067), (0.18142937, 0.54681134)] # hide
+
+inners = [[(0.87789994, 0.32551613), (0.5614043, 0.540334), (0.9494598, 0.39622766), (0.87789994, 0.32551613)], [(0.2799388, 0.52516246), (0.38555774, 0.32233855), (0.36943135, 0.30108362), (0.2799388, 0.52516246)]] # hide
+
+polyarea = PolyArea(outer, inners) # hide
+
+mesh = discretize(polyarea, FIST()) # hide
+
+plot(plot(polyarea), plot(mesh)) # hide
```
Comment (juliohm) [inline docs/src/algorithms/discretization.md]:
```suggestion
outer = [(0.18142937, 0.54681134), (0.38282228, 0.107781954), (0.43220532, 0.013640274), (0.48068276, 0.019459315), (0.48322055, 0.11583236), (0.46696007, 0.2230227), (0.48184678, 0.2656454), (0.45998818, 0.2784367), (0.4168235, 0.2190962), (0.4124987, 0.21208182), (0.39593673, 0.2520411), (0.44333926, 0.28375763), (0.4978224, 0.3981428), (0.7703431, 0.20181546), (0.7612364, 0.33008572), (0.9856581, 0.2215304), (0.99374324, 0.3353423), (0.9688778, 0.38663587), (0.59554976, 0.655444), (0.59496254, 0.58492756), (0.27641845, 0.656314), (0.3242084, 0.6072907), (0.42408508, 0.49353212), (0.20984341, 0.59003067), (0.18142937, 0.54681134)]

inners = [[(0.87789994, 0.32551613), (0.5614043, 0.540334), (0.9494598, 0.39622766), (0.87789994, 0.32551613)], [(0.2799388, 0.52516246), (0.38555774, 0.32233855), (0.36943135, 0.30108362), (0.2799388, 0.52516246)]]

polyarea = PolyArea(outer, inners)

mesh = discretize(polyarea, FIST())

plot(plot(polyarea), plot(mesh))
```

**Review by juliohm — APPROVED** [review]

**Pull request merged**